extensions/ql-vscode/src/pure/result-keys.ts

```diff
@@ -64,7 +79,7 @@ export function getPathNode(sarif: sarif.Log, key: PathNode): sarif.Location | u
 /**
  * Returns true if the two keys are both `undefined` or contain the same set of indices.
  */
-export function equals(key1: PathNode | undefined, key2: PathNode | undefined): boolean {
+export function equals(key1: Partial<PathNode> | undefined, key2: Partial<PathNode> | undefined): boolean {
```

Comment:
I don't think this function is actually used.

Reply:
Same with `getPathNode` and `getPath` above.

Reply:
After some updates, `getPathNode` is actually used now, but I removed the other unused parts.